app/Main.hs

```diff
@@ -169,22 +145,24 @@ parseValidate Config{..} = do
     d''' <- case confOutputInput of
          "" -> return d''
          x  ->  return $ progMirrorInputRelations d'' x
+    when confDumpFlat $
+        writeFile (replaceExtension confDatalogFile ".flat.ast") (show d''')
     d'''' <- case validate d''' of
                Left e   -> errorWithoutStackTrace $ "error: " ++ e
                Right d'''' -> return d''''
+    when confDumpValid $
+        writeFile (replaceExtension confDatalogFile ".valid.ast") (show d'''')
     d' <- case confDebugHooks of
          False -> return d''''
          True  -> return $ progInjectDebuggingHooks d''''
+    when confDumpDebug $
+        writeFile (replaceExtension confDatalogFile ".debug.ast") (show d')
```

Comment:
Should we add check to only dump when confDebugHooks is enabled?

Or maybe update help message that if pp-debug is passed without "-g" then this will dump the source but no debug hooks?

Reply:
Yeah, I was being lazy there. Let's go with the second approach, as it is more permissive, i.e., works better with automation.